frontend/src/components/auth/PermissionList.vue

```diff
@@ -1,387 +1,445 @@
+    return [{ value: "Schema" }];
+  }
+  if (/^ENT_[CD]/.test(newPermData.value.permission)) {
+    return [{ value: "Schema" }];
+  }
+  if (/^ENT_[UR]/.test(newPermData.value.permission)) {
+    return [{ value: "Entity" }, { value: "Schema" }];
+  }
+  return [
+    nochoice,
+    { value: "Group" },
+    { value: "Entity" },
+    { value: "Schema" },
+  ];
+});
+
+const objectIdOptions = computed(() => {
+  if (newPermData.value.objType === "Group") {
+    return Object.values(availableGroups.value).map((x) => {
+      return { value: x.id, text: x.name };
+    });
+  }
+  if (newPermData.value.objType === "Schema") {
+    return availableSchemas.value.map((x) => {
+      return { value: x.id, text: x.name };
+    });
+  }
+  return [];
+});
+
+function isPermissionInherited(perm) {
+  if (
+    props.recipientType &&
+    props.recipientType === perm.recipient_type &&
+    recipientName.value === perm.recipient_name
+  ) {
+    return false;
+  }
+  if (
+    props.objectType &&
+    props.objectType === perm.obj_type &&
+    props.objectId === perm.obj_id
+  ) {
+    return false;
+  }
+  return true;
+}
+
+const loading = ref(false);
+async function load() {
+  loading.value = true;
+  if (
+    (props.recipientType && props.recipientId) ||
+    (props.objectType && props.objectId)
+  ) {
+    permissions.value = await api.getPermissions({
+      recipientType: props.recipientType,
+      recipientId: props.recipientId,
+      objType: props.objectType,
+      objId: props.objectId,
+    });
+  } else {
+    permissions.value = [];
+  }
+  selected.value = [];
+  loading.value = false;
+}
+
+function getObjectName(perm) {
+  if (perm.obj_type === "Group") {
+    // Deleting a group doesn't cascade delete related permissions
+    // -> hence the group no longer exists, but the permissions do.
+    if (availableGroups.value[perm.obj_id]) {
+      return availableGroups.value[perm.obj_id].name;
     }
+
+    return "N/A";
+  }
+
+  if (perm.obj_type === "User") {
+    return availableGroups.value[perm.obj_id].username;
   }
+
+  if (perm.obj_type === "Schema") {
+    return availableSchemas.value
+      .filter((x) => x.id === perm.obj_id)
+      .map((x) => x.name)[0];
+  }
+  return perm.obj_id;
 }
-</script>
 
-<style scoped>
+function getObjectPermission(perm) {
+  return permOptions
+    .filter((x) => x.value === perm.permission)
+    .map((x) => x.text)[0];
+}
+
+function onSelect() {
+  const elems = document.getElementsByName("PermissionSelect");
+  selected.value = Array.prototype.filter
+    .call(elems, (e) => e.checked)
+    .map((e) => e.value);
+}
 
-</style>
\ No newline at end of file
+async function onDelete() {
+  await api.revokePermissions({
+    permissionIds: selected.value.map((x) => parseInt(x)),
+  });
+  await load();
+}
+
+async function onAddition() {
+  await api.grantPermission(newPermData.value);
+  await load();
+}
+
+await Promise.all([loadGroupData(), loadUserData()]);
```

Comment:
Do we expect the user and group data to change so frequently on the server that we need to get it every time this component gets activated?

Reply:
same as above